homeassistant/components/airvisual_pro/__init__.py

```diff
@@ -0,0 +1,106 @@
+"""The AirVisual Pro integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from pyairvisual import NodeSamba
+from pyairvisual.node import NodeProError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN, LOGGER
+
+PLATFORMS = [Platform.SENSOR]
+
+DEFAULT_UPDATE_INTERVAL = timedelta(minutes=1)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up AirVisual Pro from a config entry."""
+
+    async def async_get_data() -> dict[str, Any]:
+        """Get data from the device."""
+        try:
+            async with NodeSamba(
+                entry.data[CONF_IP_ADDRESS], entry.data[CONF_PASSWORD]
+            ) as node:
+                return await node.async_get_latest_measurements()
+        except NodeProError as err:
+            raise UpdateFailed(f"Error while retrieving data: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        LOGGER,
+        name="Node/Pro data",
+        update_interval=DEFAULT_UPDATE_INTERVAL,
+        update_method=async_get_data,
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = coordinator
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```

trivial but it does safe a line